lib/rouge/lexers/php.rb

```diff
@@ -44,6 +44,9 @@ def builtins
         end
       end
 
+      LNUM = '\d+(?:_\d+)*'
+      DNUM = '(?:(?:#{LNUM}?\.#{LNUM})|(?:#{LNUM}\.#{LNUM}?))'
+      WHITESPACE = '[ \n\r\t]+'
```

Comment:
It's not true for all the lexers in the library, but as we make changes, we try to avoid creating new objects like this. These constants will never be garbage collected. Could you inline them where they're used? I realise this makes things more difficult to read.

Reply:
Oh, and you can use `\s` to match against whitespace generally. This matches more characters than you've got listed here but I don't imagine that's going to be an issue in practice. It's a non-goal of Rouge to check syntactic correctness.

Reply:
It was intended for readability but ok.

WHITESPACE strictly matches PHP's lexer: `\s` includes `\f` and `\v`.

Reply:
Fair enough. An alternative is to create a state that you can mix in wherever whitespace is required. You can see how this can be done by looking at the JSON lexer:

https://github.com/rouge-ruby/rouge/blob/7135f0b52edcf2bf83a4c484137c130257d1ac28/lib/rouge/lexers/json.rb#L15-L20